cert/sign.go

```diff
@@ -42,22 +42,58 @@ type beingSignedCertificate interface {
 	setSignature([]byte) error
 }
 
+type SignerLambda func(certBytes []byte) ([]byte, error)
+
 // Sign will create a sealed certificate using details provided by the TBSCertificate as long as those
 // details do not violate constraints of the signing certificate.
 // If the TBSCertificate is a CA then signer must be nil.
-func (t *TBSCertificate) Sign(signer Certificate, curve Curve, key []byte) (Certificate, error) {
-	return t.sign(signer, curve, key, nil)
+func (t *TBSCertificate) Sign(signWith Certificate, curve Curve, key []byte) (Certificate, error) {
```

Comment:
I would choose whichever one reads better, `signWith` or `signer` and normalize all our signing entry points to that.